README.md

```diff
@@ -1,6 +1,261 @@
-# Flexible Streamer for SNAX
+# Flexible Streamer Generator for SNAX
+
+The Flexible Streamer Generator is built to alleviate the data layout concern of an accelerator. It decouples the the accelerator with the real memory system. It simplifies the data interface of the accelerator. The accelerator doesn't need to consider the complicated request and response to/from the real memory system. The accelerator's data interface is simply a valid-ready decoupled handshake interface. Besides, the pre-fetch mechanism inside the Streamer also helps to relieve the data contention problem.
```

Comment:
This section and all of these explanations are rather complicated, can we make it a bit "dumber" in the beginning and delay the complex stuff a bit past the intro?
Something like:

The streamer converts a programmable access pattern to a sequence of addresses in space and time to stream data to and from a banked memory system.

And maybe then lead with your SIMD example @xiaoling-yi ?